packages/aws-cdk-lib/aws-lambda-nodejs/test/bundling.test.ts

```diff
@@ -882,6 +882,23 @@ test('bundling with <= Node16 warns when sdk v3 is external', () => {
   );
 });
 
+test('bundling with <= Node16 does not warn with default externalModules', () => {
+  Bundling.bundle(stack, {
+    entry,
+    projectRoot,
+    depsLockFilePath,
+    runtime: Runtime.NODEJS_16_X,
+    architecture: Architecture.X86_64,
+  });
+
+  Annotations.fromStack(stack).hasNoWarning('*',
+    'If you are relying on AWS SDK v3 to be present in the Lambda environment already, please explicitly configure a NodeJS runtime of Node 18 or higher. [ack: @aws-cdk/aws-lambda-nodejs:sdkV3NotInRuntime]',
+  );
+  Annotations.fromStack(stack).hasNoWarning('*',
+    'If you are relying on AWS SDK v2 to be present in the Lambda environment already, please explicitly configure a NodeJS runtime of Node 16 or lower. [ack: @aws-cdk/aws-lambda-nodejs:sdkV2NotInRuntime]',
+  );
```

Comment:
Check for "does not produce a warning" this way seems very brittle, but I couldn't see a better way from looking at https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.assertions.Annotations.html.